lib/puppetserver/ca/action/prune.rb

```diff
@@ -187,6 +196,34 @@ def prune_using_certname(crl, key, inventory_file, cadir, certnames)
           end
         end
 
+        def prune_expired (crl, key, inventory_file, cadir)
+          serialnumbers = []
+          signed_dir = "#{cadir}/signed"
+          @logger.debug("Checking inventory file #{inventory_file} for expired entries") if @logger.debug?
+          errors = FileSystem.validate_file_paths(inventory_file)
+          if errors.empty?
+            File.open(inventory_file).each_line do |line|
+              if line.match(/\/CN=.*$/) && line.split.length == 4
+                not_after = line.split[2]
+                if Time.strptime(not_after,"%Y-%m-%dT%H:%M:%SUTC") && Time.parse(not_after) < Time.now
+                  serialnumbers.push(line.split.first)
+                end
+              end
+            end
+          else
+            @logger.warn "Reading inventory file at #{inventory_file} failed with error #{errors}"
+          end
+          @logger.debug("Checking CA dir #{cadir} for expired certs")
+          Dir.foreach(signed_dir) do |filename|
```

Comment:
Might be able to compact this into `Dir.glob("#{signed_dir}/*.pem").each do |filename|`

Reply:
I read somewhere Dir.foreach is preferred over Dir.glob for performance if there are huge number of files.

Reply:
Here https://stackoverflow.com/questions/2512254/iterate-through-every-file-in-one-directory

Reply:
Ah, TIL!  Makes sense, though.